### PR TITLE
feat: Add Experiment Filters for Invitations

### PR DESCRIPTION
**Pull request by leostera**
Still working here, opening to gather notes. So far:

- [x] New `hardcoded` field for Invitations
- [x] Adds a subquery for listing invitations
- [x] Creates a temporary table for Invitations
- [x] ~Add support for Assignments~ – we will add assignment filtering in the next PR
- [x] Create integration test for different filters

Like #256 this PR builds on to of #257, so once #257 is merged, I'll clean up the history here.

### CODE REVIEW

File: pool/app/experiment/entity.ml [hunk 1]
```diff
@@ -143,6 +143,31 @@ let create
     }
 ;;
 
+let id t = t.id
```
Comment (leostera) [inline pool/app/experiment/entity.ml:146]: Remove these.

File: pool/app/filter/entity.ml [hunk 2]
```diff
@@ -40,6 +40,18 @@ type value =
   | Lst of single_val list [@printer print "list"] [@name "list"]
 [@@deriving show { with_path = false }, eq, variants]
 
+module Value = struct
```
Comment (leostera) [inline pool/app/filter/entity.ml:43]: Remove these.

File: pool/app/filter/repo/repo.ml [hunk 3]
```diff
@@ -319,14 +368,18 @@ module Sql = struct
     let Dynparam.Pack (pt, pv), prepared_request =
       sql |> where_prefix |> find_filtered_request_sql ?limit use_case dyn
     in
+    Format.fprintf
```
Comment (leostera) [inline pool/app/filter/repo/repo.ml:371]: Remove these.

File: pool/test/experiment_filter_test.ml [hunk 4]
```diff
@@ -0,0 +1,221 @@
+let ( let@ ) = Result.bind
```
Comment (leostera) [inline pool/test/experiment_filter_test.ml:1]: Rename to `filter_invitation_test.ml`

File: pool/test/assignments_filter_test.ml [hunk 5]
```diff
@@ -0,0 +1,48 @@
+(* test 1: 1 assigned contact, 1 probe contact test 2: 1 assigned contact marked
```
Comment (leostera) [inline pool/test/assignments_filter_test.ml:1]: Move to next PR.

**Comment by leostera** [comment]
@mabiede feel free to squash and remove some of the commit messages in the process :)

**Review by mabiede — CHANGES_REQUESTED** [review]
Nice, thanks. Added a few comments about left overs and adjustments to the consistency

**Review by mabiede — APPROVED** [review]

**Pull request merged**